clients/cmd/fluentd/lib/fluent/plugin/out_loki.rb

```diff
@@ -313,6 +316,20 @@ def line_to_loki(record)
               end
             end
           end
+          
+          # In combination with the fluentd kubernetes_metadata plugin, 
+          # you can get namespace labels as well.
+          if @extract_kubernetes_namespace_labels && record.key?('kubernetes')
+            kubernetes_ns_labels = record['kubernetes']['namespace_labels']
+            if !kubernetes_ns_labels.nil?
+              kubernetes_ns_labels.each_key do |l|
+                new_key = l.gsub(%r{[.\-\/]}, '_')
+                # I'm not sure that a label that long is a good idea. 
```

Comment:
What is your concern regarding the length of the label?
Also, I think we can drop `_label` from the name since regular Kubernetes labels don't seem to have `label` in their name from what I can see [here](https://github.com/grafana/loki/pull/3784/files#diff-4efada4d1ebcdf1c959ffca973151b80ff553f55cca3d138d9d3ccb542de75e2R315)

Reply:
Hi, 
My concerns are purely cosmetical. 1st, such a long name looks bad on grafana UI because it gets word-wrapped. 2nd, the pods labels are not named "labels_*", so it creates a difference in the naming convention.

I agree with the idea of dropping the _label part.

I know that changing the naming of the pod labels is a breaking change, but it would be consistent with the "namespace_..." name for namespace labels.

In our fluentd+graylog current setup, as an example, labels are : 
* labels_something for the pod label something
* namespace_labels_something, for the namespace label something.

I don't have a real strong opinion about all that. As you said, @dannykopping , maybe the others will have something to say.